From a84b60f679add735c4640435cb29edb5066b97d0 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Fri, 07 Jun 2019 22:30:23 +0300
Subject: [PATCH] [SW-4185] gaudi: pll: raise pci lbw clock

The PCI PLL is different from others as it supplies clocks to
PCI, PSOC, SPI, etc. and needs special care while switching it.
We switch the PCI PLL HBW and the LBW clocks in the BTL which includes
gating them, but we can't do the same for DIV2 output.
Moreover, DIV2 and DIV3 outputs do not need any W/A to be switched as
their frequency is below 400MHz.
Therefore, we do the LBW switch in BTL and just switch (no gating) of
the DIV2,3 in Preboot.
For this we use the special HL_PLL_WA_NOGT enum value, so no gating will
be done during the switch.

Change-Id: Ifc539af73a30e724a1a313b69d6551582666d592
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/asic.c b/gaudi/asic.c
index c09eff5..ab00579 100644
--- a/gaudi/asic.c
+++ b/gaudi/asic.c
@@ -148,14 +148,14 @@
 		.nr = 4, .nf = 331, .od = 3, .nb = 165, //DAY 1 (830MHz)
 		.div_factor = { 0, 0, 4, 9 },
 		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK
+			     HL_PLL_PLL_CLK,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL
 		},
 		.div_wa = { HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA
+			    HL_PLL_WA_NOGT,
+			    HL_PLL_WA_NOGT
 		},
 	}, {
 		.name = "TPC", .base = mmPSOC_TPC_PLL_BASE,
@@ -395,7 +395,6 @@
 	for (i = 0; i < ARRAY_SIZE(gaudi_pll_cfg); i++) {
 		switch (gaudi_pll_cfg[i].base) {
 		case mmPSOC_CPU_PLL_BASE:
-		case mmPSOC_PCI_PLL_BASE:
 			continue;
 		case mmNIC2_HBM_PLL_BASE:
 			if (gaudi_reset_soc_done()) {
